src/main/java/com/squareup/kotlinpoet/CodeBlock.kt

```diff
@@ -150,14 +150,18 @@ class CodeBlock private constructor(
   override fun equals(other: Any?): Boolean {
     if (this === other) return true
     if (other == null) return false
-    if (javaClass != other.javaClass) return false
-    return toString() == other.toString()
+    if (other !is CodeBlock) return false
+    return toUnescapedString() == other.toUnescapedString()
   }
 
-  override fun hashCode() = toString().hashCode()
+  override fun hashCode() = toUnescapedString().hashCode()
 
   override fun toString() = buildString { CodeWriter(this).emitCode(this@CodeBlock) }
 
+  private fun toUnescapedString() = toString().run {
+    if (isNotEmpty() && first() == '`' && last() == '`') substring(1, length - 1) else this
```

Comment:
The `first()` and `last()` look good in context of the bug. Just feels like `toUnescapedString()` is a misnomer since it only unescapes examples like "`when`" and not examples where the escaping might be somewhere in the middle of the CodeBlock (of course those examples wouldn't need the `equals()` then)